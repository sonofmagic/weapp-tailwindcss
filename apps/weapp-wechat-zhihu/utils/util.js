function formatTime(date) {
  const year = date.getFullYear()
  const month = date.getMonth() + 1
  const day = date.getDate()

  const hour = date.getHours()
  const minute = date.getMinutes()
  const second = date.getSeconds()

  return `${[year, month, day].map(formatNumber).join('/')} ${[hour, minute, second].map(formatNumber).join(':')}`
}

function formatNumber(n) {
  n = n.toString()
  return n[1] ? n : `0${n}`
}

module.exports = {
  formatTime,
}

const index = require('../data/data_index.js')
const index_next = require('../data/data_index_next.js')
const discovery = require('../data/data_discovery.js')
const discovery_next = require('../data/data_discovery_next.js')

function getData(url) {
  return new Promise((resolve, reject) => {
    wx.request({
      url,
      data: {},
      header: {
        // 'Content-Type': 'application/json'
      },
      success(res) {
        console.log('success')
        resolve(res)
      },
      fail(res) {
        reject(res)
        console.log('failed')
      },
    })
  })
}

function getData2() {
  return index.index
}

function getNext() {
  return index_next.next
}

function getDiscovery() {
  return discovery.discovery
}

function discoveryNext() {
  return discovery_next.next
}

module.exports.getData = getData
module.exports.getData2 = getData2
module.exports.getNext = getNext
module.exports.getDiscovery = getDiscovery
module.exports.discoveryNext = discoveryNext
