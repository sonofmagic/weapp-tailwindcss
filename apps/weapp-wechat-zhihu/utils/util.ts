import index from '@/data/data_index.js'
import index_next from '@/data/data_index_next.js'
import discovery from '@/data/data_discovery.js'
import discovery_next from '@/data/data_discovery_next.js'

function formatTime(date) {
  const year = date.getFullYear()
  const month = date.getMonth() + 1
  const day = date.getDate()

  const hour = date.getHours()
  const minute = date.getMinutes()
  const second = date.getSeconds()

  return `${[year, month, day].map(formatNumber).join('/')} ${[hour, minute, second].map(formatNumber).join(':')}`
}

function formatNumber(n) {
  n = n.toString()
  return n[1] ? n : `0${n}`
}

function getData(url) {
  return new Promise((resolve, reject) => {
    wx.request({
      url,
      data: {},
      header: {
        // 'Content-Type': 'application/json'
      },
      success(res) {
        console.log('success')
        resolve(res)
      },
      fail(res) {
        reject(res)
        console.log('failed')
      },
    })
  })
}

function getData2() {
  return index.index
}

function getNext() {
  return index_next.next
}

function getDiscovery() {
  return discovery.discovery
}

function discoveryNext() {
  return discovery_next.next
}

export {
  getData,
  getData2,
  getNext,
  getDiscovery,
  discoveryNext,
  formatTime,
}
