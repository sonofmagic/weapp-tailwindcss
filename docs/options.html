<!doctype html>
<html lang="zh-cn" dir="ltr" class="docs-wrapper plugin-docs plugin-id-default docs-version-current docs-doc-page docs-doc-id-options/index" data-has-hydrated="false">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v3.7.0">
<title data-rh="true">配置项(v3.x) | weapp-tailwindcss 把tailwindcss带给小程序开发者们</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:image" content="https://tw.icebreaker.top/weapp-tailwindcss/img/logo.png"><meta data-rh="true" name="twitter:image" content="https://tw.icebreaker.top/weapp-tailwindcss/img/logo.png"><meta data-rh="true" property="og:url" content="https://tw.icebreaker.top/weapp-tailwindcss/docs/options"><meta data-rh="true" property="og:locale" content="zh_cn"><meta data-rh="true" name="docusaurus_locale" content="zh-cn"><meta data-rh="true" name="docsearch:language" content="zh-cn"><meta data-rh="true" name="keywords" content="weapp,小程序,tailwindcss,原子类,uni-app,taro,rax,mpx,native,remax,原生,webpack,plugin,vite,gulp,wxss,wxml"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="配置项(v3.x) | weapp-tailwindcss 把tailwindcss带给小程序开发者们"><meta data-rh="true" name="description" content="目前所有的配置项文档已经通过 typedoc 来自动生成."><meta data-rh="true" property="og:description" content="目前所有的配置项文档已经通过 typedoc 来自动生成."><link data-rh="true" rel="icon" href="/weapp-tailwindcss/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://tw.icebreaker.top/weapp-tailwindcss/docs/options"><link data-rh="true" rel="alternate" href="https://tw.icebreaker.top/weapp-tailwindcss/docs/options" hreflang="zh-cn"><link data-rh="true" rel="alternate" href="https://tw.icebreaker.top/weapp-tailwindcss/docs/options" hreflang="x-default"><link data-rh="true" rel="preconnect" href="https://9Y7BJULSEW-dsn.algolia.net" crossorigin="anonymous"><link rel="preconnect" href="https://www.google-analytics.com">
<link rel="preconnect" href="https://www.googletagmanager.com">
<script async src="https://www.googletagmanager.com/gtag/js?id=G-S81Q4GRTPM"></script>
<script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-S81Q4GRTPM",{})</script>



<link rel="search" type="application/opensearchdescription+xml" title="weapp-tailwindcss 把tailwindcss带给小程序开发者们" href="/weapp-tailwindcss/opensearch.xml">



<link rel="preconnect" href="https://hm.baidu.com">
<meta name="baidu-site-verification" content="codeva-4ny6UzMmrn">
<script src="https://hm.baidu.com/hm.js?61f3de7065e36044e6d5f201632bc368" async></script><link rel="stylesheet" href="/weapp-tailwindcss/assets/css/styles.5e9a1158.css">
<script src="/weapp-tailwindcss/assets/js/runtime~main.99a836e5.js" defer="defer"></script>
<script src="/weapp-tailwindcss/assets/js/main.2dab0032.js" defer="defer"></script>
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){try{return new URLSearchParams(window.location.search).get("docusaurus-theme")}catch(t){}}()||function(){try{return window.localStorage.getItem("theme")}catch(t){}}();t(null!==e?e:"dark")}(),function(){try{const n=new URLSearchParams(window.location.search).entries();for(var[t,e]of n)if(t.startsWith("docusaurus-data-")){var a=t.replace("docusaurus-data-","data-");document.documentElement.setAttribute(a,e)}}catch(t){}}()</script><div id="__docusaurus"><link rel="preload" as="image" href="/weapp-tailwindcss/img/logo.png"><div class="relative"><div role="region" aria-label="跳到主要内容"><a class="skipToContent_K2qX" href="#__docusaurus_skipToContent_fallback">跳到主要内容</a></div><nav aria-label="主导航" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="切换导航栏" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/weapp-tailwindcss/"><div class="navbar__logo"><img src="/weapp-tailwindcss/img/logo.png" alt="weapp tailwindcss Logo" class="themedComponent_MInF themedComponent--light_S2gH"><img src="/weapp-tailwindcss/img/logo.png" alt="weapp tailwindcss Logo" class="themedComponent_MInF themedComponent--dark_T1fd"></div><b class="navbar__title text--truncate">weapp-tailwindcss</b></a><a class="navbar__item navbar__link" href="/weapp-tailwindcss/docs/intro">指南</a><a class="navbar__item navbar__link" href="/weapp-tailwindcss/docs/community/templates">生态及解决方案</a><a class="navbar__item navbar__link" href="/weapp-tailwindcss/docs/issues">常见问题</a><a class="navbar__item navbar__link" href="/weapp-tailwindcss/docs/showcase">优秀案例展示</a><a class="navbar__item navbar__link" href="/weapp-tailwindcss/docs/migrations/v3">更新与迁移</a><a class="navbar__item navbar__link" href="/weapp-tailwindcss/docs/api/interfaces/UserDefinedOptions">配置项</a><a href="https://vite.icebreaker.top/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">🔥Weapp-vite<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><a href="https://icebreaker.top/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">博客<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/sonofmagic/weapp-tailwindcss" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_KkYb colorModeToggle_RQmm"><button class="clean-btn toggleButton_IrMR toggleButtonDisabled_CZnP" type="button" disabled="" title="切换浅色/暗黑模式（当前为暗黑模式）" aria-label="切换浅色/暗黑模式（当前为暗黑模式）" aria-live="polite" aria-pressed="true"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_NE0E"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_BX3z"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="navbarSearchContainer_nREN"><button type="button" class="DocSearch DocSearch-Button" aria-label="搜索 (Command+K)"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20" aria-hidden="true"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">搜索</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="__docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_oXQn"><div class="docsWrapper__vBA"><button aria-label="回到顶部" class="clean-btn theme-back-to-top-button backToTopButton_cUbq" type="button"></button><div class="docRoot_YSGh"><main class="docMainContainer_rxAt docMainContainerEnhanced_UfzA"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_eYY2"><div class="docItemContainer_mTq3"><article><div class="theme-doc-markdown markdown"><header><h1>配置项(v3.x)</h1></header>
<div class="theme-admonition theme-admonition-info admonition_IzYv alert alert--info"><div class="admonitionHeading_a_m_"><span class="admonitionIcon_t1pO"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg></span>信息</div><div class="admonitionContent_FuNW"><p>目前所有的配置项文档已经通过 <code>typedoc</code> 来自动生成.</p><p><a href="/weapp-tailwindcss/docs/api/interfaces/UserDefinedOptions">点击跳转到用户配置项: <code>UserDefinedOptions</code></a></p></div></div></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="row margin-top--sm theme-doc-footer-edit-meta-row"><div class="col"><a href="https://github.com/sonofmagic/weapp-tailwindcss/tree/main/website/docs/options/index.md" target="_blank" rel="noopener noreferrer" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_PKKU" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>编辑此页</a></div><div class="col lastUpdated_G2Ms"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="文件选项卡"></nav></div></div></div></div></main></div></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">文档</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/weapp-tailwindcss/docs/intro">指南</a></li><li class="footer__item"><a class="footer__link-item" href="/weapp-tailwindcss/docs/options">配置项</a></li><li class="footer__item"><a class="footer__link-item" href="/weapp-tailwindcss/docs/issues">常见问题</a></li></ul></div><div class="col footer__col"><div class="footer__title">更多</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/sonofmagic/weapp-tailwindcss" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/sonofmagic/weapp-tailwindcss/blob/main/CODE_OF_CONDUCT.md" target="_blank" rel="noopener noreferrer" class="footer__link-item">Code of Conduct<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://vite.icebreaker.top" target="_blank" rel="noopener noreferrer" class="footer__link-item">weapp-vite<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://icebreaker.top" target="_blank" rel="noopener noreferrer" class="footer__link-item">博客<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://ui.icebreaker.top/zh-CN" target="_blank" rel="noopener noreferrer" class="footer__link-item">IceStack<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/sonofmagic/weapp-pandacss" target="_blank" rel="noopener noreferrer" class="footer__link-item">weapp-pandacss<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_oyey"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright"><div class="flex flex-col items-center justify-center space-y-2">
        <span><a target="_self" rel="nofollow" href="/docs/copyright">Copyright</a> © 2025 <a target="_blank" rel="nofollow" href="https://github.com/sonofmagic">sonofmagic</a></span>
        <span class="flex items-center">This site is powered by <a target="_blank" rel="nofollow" href="https://www.netlify.com"><img class="w-10 ml-2" src="/img/logo-netlify.png" alt="Netlify badge"></a></span>
        </div></div></div></div></footer><div class="pointer-events-none absolute inset-0 z-[202] flex-none"><div class="size-full rounded-none bg-[url(/img/framer.png)] bg-[length:128px] bg-repeat opacity-5"></div></div></div></div>
</body>
</html>