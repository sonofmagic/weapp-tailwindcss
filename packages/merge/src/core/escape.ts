import type { CreateOptions, EscapeFn } from '../types'
import { escape } from 'weapp-tailwindcss/escape'

const noop = (value: string) => value

export function resolveEscape(options?: CreateOptions): EscapeFn {
  if (options?.disableEscape) {
    return noop
  }

  return options?.escapeFn ?? escape
}

export {
  noop,
}
