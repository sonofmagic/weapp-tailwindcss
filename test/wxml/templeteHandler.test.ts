import { format, wxmlCasePath, createGetCase } from '../util'
import { templateHandler } from '#test/v2/wxml'
import { MappingChars2String } from '@/escape'
import { getOptions } from '@/options'

const getWxmlCase = createGetCase(wxmlCasePath)

function complexHandler(str: string) {
  return templateHandler(str, {
    escapeMap: MappingChars2String
  })
}

describe('virtualHostClass', () => {
  // 开启mergeVirtualHostAttributes
  it('virtualHostClass case 0', async () => {
    const testCase = await getWxmlCase('virtualHost-case0.wxml')
    const { templateHandler } = getOptions()
    const str = templateHandler(testCase)
    expect(str).toMatchSnapshot()
  })

  // 不开启mergeVirtualHostAttributes
  it('virtualHostClass case 1', async () => {
    const testCase = await getWxmlCase('virtualHost-case1.wxml')
    const { templateHandler } = getOptions()
    const str = templateHandler(testCase)
    expect(str).toMatchSnapshot()
  })

  it('mpx after content double qutos', () => {
    const x = '<view class="after:content-["你好啊，我很无聊"] after:ml-0.5 after:text-red-500"></view>'
    const { templateHandler } = getOptions()
    expect(templateHandler(x)).toMatchSnapshot()
  })
})

describe.skip('templateHandler', () => {
  test('wildcard char', () => {
    const testCase = "<view class=\"{{['flex','flex-col','items-center',flag===1?'bg-red-900':'bg-[#fafa00]']}}\">*****</view>"

    const str = complexHandler(testCase)
    expect(str).toBe("<view class=\"{{['flex','flex-col','items-center',flag===1?'bg-red-900':'bg-_bl__h_fafa00_br_']}}\">*****</view>")
  })

  test('only wildcard char', () => {
    const testCase = '<view>*****</view>'

    const str = templateHandler(testCase)
    expect(str).toBe('<view>*****</view>')
  })
  test('with var', () => {
    const testCase = "<view class=\"{{['flex','flex-col','items-center',flag===1?'bg-red-900':'bg-[#fafa00]']}}\"></view>"

    const str = complexHandler(testCase)
    expect(str).toBe("<view class=\"{{['flex','flex-col','items-center',flag===1?'bg-red-900':'bg-_bl__h_fafa00_br_']}}\"></view>")
  })

  test('dark mode and hover-class', () => {
    const testCase = '<view class="bg-gray-100 dark:bg-zinc-800" hover-class="bg-red-500 dark:bg-green-500"></view>'
    const str = complexHandler(testCase)
    expect(str).toBe('<view class="bg-gray-100 dark_c_bg-zinc-800" hover-class="bg-red-500 dark_c_bg-green-500"></view>')
  })

  it('wxs should be ignored ', () => {
    const testCase = `<wxs module="status">
    function get(index, active) {
      if (index < active) {
        return 'finish';
      } else if (index === active) {
        return 'process';
      }

      return 'inactive';
    }

    module.exports = get;
    </wxs>`
    const result = complexHandler(testCase)

    expect(result).toBe(testCase)
  })

  it('should ', () => {
    const testCase = `<button class="
    u-reset-button
    rounded-full
    w-10
    h-10
    flex
    justify-center
    items-center
    pointer-events-auto
    bg-white
    shadow
  " open-type="contact"><tui-icon vue-id="{{('17a41c02-4')+','+('17a41c02-3')}}" name="kefu" color="rgb(41, 121, 255)" size="40" unit="rpx" bind:__l="__l"></tui-icon></button>`

    const result = complexHandler(testCase)

    expect(result).toBe(
      '<button class="    u-reset-button    rounded-full    w-10    h-10    flex    justify-center    items-center    pointer-events-auto    bg-white    shadow  " open-type="contact"><tui-icon vue-id="{{(\'17a41c02-4\')+\',\'+(\'17a41c02-3\')}}" name="kefu" color="rgb(41, 121, 255)" size="40" unit="rpx" bind:__l="__l"></tui-icon></button>'
    )
  })

  test('with var 3', () => {
    const testCase = "<view class=\"{{[flag<2?'a':'b',flag>=1?'bg-red-900':'bg-[#fafa00]']}}\"></view>"
    const res = complexHandler(testCase)
    expect(res).toBe("<view class=\"{{[flag<2?'a':'b',flag>=1?'bg-red-900':'bg-_bl__h_fafa00_br_']}}\"></view>")
  })

  test('with var 4', () => {
    const testCase = `<button id="{{ id }}" data-detail="{{ dataset }}" class="custom-class {{ utils.bem('button', [type, size, { block, round, plain, square, loading, disabled, hairline, unclickable: disabled || loading }]) }} {{ hairline ? 'van-hairline--surround' : '' }}" hover-class="van-button--active hover-class" lang="{{ lang }}" form-type="{{ formType }}" style="{{ computed.rootStyle({ plain, color, customStyle }) }}" open-type="{{ disabled || loading || (canIUseGetUserProfile && openType === 'getUserInfo') ? '' : openType }}" business-id="{{ businessId }}" session-from="{{ sessionFrom }}" send-message-title="{{ sendMessageTitle }}" send-message-path="{{ sendMessagePath }}" send-message-img="{{ sendMessageImg }}" show-message-card="{{ showMessageCard }}" app-parameter="{{ appParameter }}" aria-label="{{ ariaLabel }}" bindtap="{{ disabled || loading ? '' : 'onClick' }}" bindgetuserinfo="onGetUserInfo" bindcontact="onContact" bindgetphonenumber="onGetPhoneNumber" binderror="onError" bindlaunchapp="onLaunchApp" bindopensetting="onOpenSetting">
    </button>`

    const res = complexHandler(testCase)
    expect(res)
      .toBe(`<button id="{{ id }}" data-detail="{{ dataset }}" class="custom-class {{utils.bem('button',[type,size,{block,round,plain,square,loading,disabled,hairline,unclickable:disabled||loading}])}} {{hairline?'van-hairline--surround':''}}" hover-class="van-button--active hover-class" lang="{{ lang }}" form-type="{{ formType }}" style="{{ computed.rootStyle({ plain, color, customStyle }) }}" open-type="{{ disabled || loading || (canIUseGetUserProfile && openType === 'getUserInfo') ? '' : openType }}" business-id="{{ businessId }}" session-from="{{ sessionFrom }}" send-message-title="{{ sendMessageTitle }}" send-message-path="{{ sendMessagePath }}" send-message-img="{{ sendMessageImg }}" show-message-card="{{ showMessageCard }}" app-parameter="{{ appParameter }}" aria-label="{{ ariaLabel }}" bindtap="{{ disabled || loading ? '' : 'onClick' }}" bindgetuserinfo="onGetUserInfo" bindcontact="onContact" bindgetphonenumber="onGetPhoneNumber" binderror="onError" bindlaunchapp="onLaunchApp" bindopensetting="onOpenSetting">
    </button>`)
  })

  test('with var 2', () => {
    const navbarTestCase = format(
      `<view data-aaa="{{aaa || 'a'}} t" disabled="true" hidden class="{{['tui-navigation-bar','data-v-ec49da2a',(opacity>0.85&&splitLine)?'tui-bar-line':'',(isFixed)?'tui-navbar-fixed':'',(backdropFilter&&dropDownOpacity>0)?'tui-backdrop__filter':'']}}" style="{{'height:'+(height+'px')+';'+('background-color:'+('rgba('+background+','+opacity+')')+';')+('opacity:'+(dropDownOpacity)+';')+('z-index:'+(isFixed?zIndex:'auto')+';')}}">
      <block wx:if="{{isImmersive}}">
          <view class="tui-status-bar data-v-ec49da2a" style="{{'height:'+(statusBarHeight+'px')+';'}}"></view>
      </block>
      <block wx:if="{{title&&!isCustom}}">
          <view class="tui-navigation_bar-title data-v-ec49da2a" style="{{'opacity:'+(transparent||opacity>=maxOpacity?1:opacity)+';'+('color:'+(color)+';')+('padding-top:'+(top-statusBarHeight+'px')+';')}}">
              {{''+title+''}}
          </view>
      </block>
      <slot></slot>
  </view>`
    )
    const str = complexHandler(navbarTestCase)
    expect(str).toBe(
      format(
        `<view data-aaa="{{aaa || 'a'}} t" disabled="true" hidden class="{{['tui-navigation-bar','data-v-ec49da2a',opacity>0.85&&splitLine?'tui-bar-line':'',isFixed?'tui-navbar-fixed':'',backdropFilter&&dropDownOpacity>0?'tui-backdrop__filter':'']}}" style="{{'height:'+(height+'px')+';'+('background-color:'+('rgba('+background+','+opacity+')')+';')+('opacity:'+(dropDownOpacity)+';')+('z-index:'+(isFixed?zIndex:'auto')+';')}}">
    <block wx:if="{{isImmersive}}">
        <view class="tui-status-bar data-v-ec49da2a" style="{{'height:'+(statusBarHeight+'px')+';'}}"></view>
    </block>
    <block wx:if="{{title&&!isCustom}}">
        <view class="tui-navigation_bar-title data-v-ec49da2a" style="{{'opacity:'+(transparent||opacity>=maxOpacity?1:opacity)+';'+('color:'+(color)+';')+('padding-top:'+(top-statusBarHeight+'px')+';')}}">
            {{''+title+''}}
        </view>
    </block>
    <slot></slot>
</view>`
      )
    )
  })

  it('class with string var', () => {
    const testCase = '<button class="btn a{{num}}" bindtap="onTap">{{num}}</button>'
    const str = complexHandler(testCase)
    expect(str).toBe(testCase)
  })

  it('utf8 test case 0', () => {
    const testCase = `<view class="{{['td',[(g.type==='你好啊')?'highlight':'']]}}">{{g.type}}</view>`
    const str = complexHandler(testCase)
    expect(str).toBe(`<view class="{{['td',[g.type==='你好啊'?'highlight':'']]}}">{{g.type}}</view>`)
  })

  it('utf8 test case 1', () => {
    const testCase = `<view class="{{[[g['人生']==='你好啊'?'highlight':'']]}}">{{g.type}}</view>`
    const str = complexHandler(testCase)
    expect(str).toBe(testCase)
  })

  it('after wx:if <view wx:if="{{xxx}}" class="ml-[16px]">', () => {
    const testCase = '<view wx:if="{{xxx}}" class="ml-[16px]">'
    const str = templateHandler(testCase)
    expect(str).toBe('<view wx:if="{{xxx}}" class="ml-_16px_">')
  })

  it('before wx:if <view  class="ml-[16px]" wx:if="{{xxx}}">', () => {
    const testCase = '<view class="ml-[16px]" wx:if="{{xxx}}">'
    const str = templateHandler(testCase)
    expect(str).toBe('<view class="ml-_16px_" wx:if="{{xxx}}">')
  })

  it('wx:if > before', () => {
    const testCase = '<view class="mt-[8px]" wx:if="{{ xxx.length > 0 }}">'
    const str = templateHandler(testCase)
    expect(str).toBe('<view class="mt-_8px_" wx:if="{{ xxx.length > 0 }}">')
  })
  it('wx:if > after', () => {
    const testCase = '<view wx:if="{{ xxx.length > 0 }}" class="mt-[8px]">'
    const str = templateHandler(testCase)
    expect(str).toBe('<view wx:if="{{ xxx.length > 0 }}" class="mt-_8px_">')
  })
})
