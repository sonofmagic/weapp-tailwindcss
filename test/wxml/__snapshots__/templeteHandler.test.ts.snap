// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`virtualHostClass > mpx after content double qutos 1`] = `"<view class=\\"afterccontent-_\\"你好啊，我很无聊\\"] after:ml-0.5 after:text-red-500\\"></view>"`;

exports[`virtualHostClass > virtualHostClass case 0 1`] = `"<view class=\\"{{[virtualHostClass]}}\\" style=\\"{{virtualHostStyle}}\\"><com-a u-s=\\"{{['d']}}\\" class=\\"text-_red_ text-_30px_\\" virtualHostClass=\\"text-_red_ text-_30px_\\" style=\\"background:#d4eb07\\" virtualHostStyle=\\"background:#d4eb07\\" u-i=\\"4c6cb2cd-0\\" bind:__l=\\"__l\\">11111</com-a></view>"`;

exports[`virtualHostClass > virtualHostClass case 1 1`] = `"<view><com-a u-s=\\"{{['d']}}\\" class=\\"text-_red_ text-_30px_\\" style=\\"background:#d4eb07\\" u-i=\\"4c6cb2cd-0\\" bind:__l=\\"__l\\">11111</com-a></view>"`;
